pkg/multi-tenancy/config/config.go

```diff
@@ -0,0 +1,37 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package config
+
+// TenantLabelKey is a label key reserved for multi-tenancy.
+const TenantLabelKey = "__tenant__"
+
+// Config defines the configuration for multi-tenancy.
+type Config struct {
+	ValidTenants []string
+	tenantsCache map[string]struct{}
+}
+
+// FillTenantCache creates a cache of valid tenants for multi-tenancy.
+func (cfg *Config) FillTenantCache() {
+	cfg.tenantsCache = make(map[string]struct{})
+	for _, tname := range cfg.ValidTenants {
+		cfg.tenantsCache[tname] = struct{}{}
+	}
+}
+
+// IsTenantAllowed returns true if the given tenantName is allowed to be ingested.
+func (cfg *Config) IsTenantAllowed(tenantName string) bool {
+	if tenantName == "" {
+		// Do not allow empty tenant names.
+		return false
+	}
+	if len(cfg.tenantsCache) == 0 {
```

Comment:
I don't get why this returns true

Reply:
if we want an allowed all mode, I'd suggest a separate bool in the config object. That way there can be no confusion

Reply:
Or better yet make this a different struct altogether that also implements `IsTenantAllowed`

Reply:
3rd comment is not clear. But going with bool as its easier to read.

Reply:
The better design would be to make an interface that is implemented by a config that allows all tenants and a config that is selective on tenants.

Reply:
done